torch/testing/_internal/common_fsdp.py

```diff
@@ -1352,6 +1355,40 @@ def _test_fsdp_parity(
             )
 
 
+def test_compiled_fsdp(compile_compute_on_module: Optional[type] = None):
```

Comment:
nit: Maybe we just `@no_type_check` this function since we are doing some patches, which mypy does not like and requires several `type: ignore` comments.

Reply:
nice! this should be cleaner. I improved patching to support ``fully_shard.state`` in an upcoming diff. Will see MYPY still complains